metadata-ingestion/src/datahub/ingestion/source/dremio/dremio_api.py

```diff
@@ -343,14 +343,147 @@ def _fetch_all_results(self, job_id: str) -> List[Dict]:
 
         while True:
             result = self.get_job_result(job_id, offset, limit)
-            rows.extend(result["rows"])
+
+            # Handle cases where API response doesn't contain 'rows' key
+            # This can happen with OOM errors or when no rows are returned
+            if "rows" not in result:
+                logger.warning(
+                    f"API response for job {job_id} missing 'rows' key. "
+                    f"Response keys: {list(result.keys())}"
+                )
+                # Check for error conditions
+                if "errorMessage" in result:
+                    raise DremioAPIException(f"Query error: {result['errorMessage']}")
+                elif "message" in result:
+                    logger.warning(
+                        f"Query warning for job {job_id}: {result['message']}"
+                    )
+                # Return empty list if no rows key and no error
+                break
+
+            # Handle empty rows response
+            result_rows = result["rows"]
+            if not result_rows:
+                logger.debug(
+                    f"No more rows returned for job {job_id} at offset {offset}"
+                )
+                break
+
+            rows.extend(result_rows)
+
+            # Check row count to determine if we should continue
+            row_count = result.get("rowCount", 0)
```

Comment:
Seems like we have 3 values, which should logically be the equal:
* `results['rowCount']`
* `len(results_rows)`
* `limit`
I think we should commit to using just one and I lean towards getting length of actual results. I see it as defensive measures to potential problems. WDYT?